lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -742,6 +749,9 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
     this._inputConfiguration = inputConfig;
 
     _domElement = inputConfig.inputType.createDomElement();
+    if (inputConfig.readOnly) {
+      domElement.setAttribute('readonly', 'readonly');
+    }
```

Comment:
This won't show the autofill suggestions anymore right?

Reply:
If the autofill suggestions are not shown, it might better not to continue the normal flow for autofill. Such as creating a form and leaving the form on the DOM.

Reply:
I don't think it makes sense for the app to have a read-only field with autofill hints, does it?

Reply:
no, it doesn't. We will add an assert to the framework side right?

Reply:
Added.